### DISCUSSION TITLE
[Question] Change in AlertDialog

### DISCUSSION THREAD

**Discussion by ingdesarr1** [flet-dev/flet#3544] [Q&A]
### Question

Good afternoon. Looking at the new planned changes, I note that the alertdialog will no longer be an element, but will have to be added with an overlay. I currently have many AlertDialogs in my program, as I use them within classes to confirm information and to fill in data. How can this change affect my process and how can I start migrating to this change?

Thanks

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment]
I dont know what exactly you read, but `AlertDialog` is still a normal Flet control. The only thing that has been changed/deprecated, is how it was opened with the use of helper methods.
Before, one could do `page.show_dialog(dlg)` to display the dialog, but now that method has been marked as deprecated in favor of the new way: `page.open(dlg)`. 
If you didnt use this method to open your dialogs (ex: you just toggle the `open` prop), then you don't have to do any migration.

However, your code should still work with the latest Flet version, since we just deprecated those methods, without yet removing them.
If you face any issues, feel free to drop them here.
